File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
from collections import defaultdict

def solution(s):
    dic = defaultdict(lambda: -1)
    answer = []
    for i in range(len(s)):
        c = s[i]
        if dic[c] > -1: answer.append(i-dic[s[i]])
        else: answer.append(-1)
        dic[s[i]] = i
    return answer

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        List<Integer> answer = new ArrayList();
        Map<Character, Integer> map = new HashMap();
        char c;
        for (int i=0;i < s.length();i++) {
            c = s.charAt(i);
            if (map.containsKey(c)) {
                answer.add(i - map.get(c));
            } else {
                answer.add(-1);
            }
            
            map.put(c, i);
        }
        
        return answer.stream().mapToInt(integer -> integer).toArray();
    }
}